pkg/build/ignore/docker/dockign.go

```diff
@@ -0,0 +1,116 @@
+package docker
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/openshift/source-to-image/pkg/api"
+)
+
+type DOCKIGN struct {
+	Config *api.Config
+}
+
+func New(req *api.Config) *DOCKIGN {
+	d := &DOCKIGN{
+		Config: req,
+	}
+	return d
+}
+
+func (b *DOCKIGN) Ignore(config *api.Config) {
```

Comment:
Passing the struct in as an arg is fine (and doing it as a ptr definitely makes sense), my primary concern here was more with the fact that you had also defined a struct with a Config field, which isn't needed since you're passing the Config object in to the only method on the struct.  Defining the struct is fine (necessary since you're doing it as an interface and not a first class function, which as you note follows the Prepare pattern and that's fine with me)

So mostly I think this comes down to:
remove the Config field from the struct
remove the New() method